front/components/database/DatabaseModal.tsx

```diff
@@ -0,0 +1,257 @@
+import { Button, Input, Modal } from "@dust-tt/sparkle";
+import { CoreAPIDatabase, DataSourceType } from "@dust-tt/types";
+import React, { useContext, useRef } from "react";
+import { useSWRConfig } from "swr";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { handleFileUploadToText } from "@app/lib/client/handle_file_upload";
+import { useDatabaseTables } from "@app/lib/swr";
+
+export function DatabaseModal({
+  isOpen,
+  onClose,
+  database,
+  dataSource,
+  workspaceId,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  database: CoreAPIDatabase | null;
+  dataSource: DataSourceType | null;
+  workspaceId: string;
+}) {
+  const { mutate } = useSWRConfig();
+
+  if (!dataSource) {
+    return null;
+  }
+
+  const { tables } = useDatabaseTables({
+    workspaceId,
+    dataSourceName: dataSource.name,
+    databaseId: database?.database_id,
+  });
+
+  const [name, setName] = React.useState("");
+  const [editedDescriptionByTableId, setEditedDescriptionByTableId] =
+    React.useState<Record<string, string>>({});
+  const [newTableDescription, setNewTableDescription] = React.useState("");
+
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const close = () => {
+    setName("");
+    setEditedDescriptionByTableId({});
+    onClose();
+  };
+
+  // Not empty, only alphanumeric, and not too long
+  const isNameValid = () => name !== "" && /^[a-zA-Z0-9_]{1,32}$/.test(name);
+
+  const sendNotification = useContext(SendNotificationsContext);
+
+  return (
+    <Modal
```

Comment:
This should be a full screen modal.

Please inspire yourself of the Document upload code/UI with titles and descriptions for each input fields.

The upload CSV button can be made bigger and should probably be centered or on the right.

Reply:
I reworked the form and followed the design from our other Doc upload inputs